ui/segmented-bar.reel/segmented-bar.js

```diff
@@ -0,0 +1,259 @@
+    prepareForActivationEvents: {
+        value: function () {
+            // more performant than this.super()
+            AbstractButton.prepareForActivationEvents.call(this);
+
+            this._defineSelectionBinding();
+        }
+    },
+
+    /**
+     * Cancel bindings for perf
+     * Ensure not to cancel bindings SegmentedBar didn't define
+     *
+     * @method
+     */
+    exitDocument: {
+        value: function () {
+            this.cancelBindings();
+
+            if (this.templateObjects.repetition.getBinding("contentController")) {
+                this.templateObjects.repetition.cancelBinding("contentController");
+            }
+            if (this.templateObjects.repetition.getBinding("content")) {
+                this.templateObjects.repetition.cancelBinding("content");
+            }
+        }
+    },
+
+    /**
+     * Hack that uses repetition selection binding to trigger dispatchActionEvent
+     * due to repetition not dispatching events on its own
+     *
+     * @private
+     * @param {Object} selectedSegmentContent
+     */
+    _selectedSegmentContent: {
+        get: Function.noop,
+        set: function (selectedSegmentContent) {
+            if (selectedSegmentContent && this.isSelectionEnabled) {
+                // Inheirted AbstractControl.detail is a Dict
+                this.detail.set("data", selectedSegmentContent);
+                this.dispatchActionEvent();
+            }
+        }
+    },
+
+    /**
+     * @private
+     * @prop {String} value - "horizontal"|"vertical"
+     * @TODO: change type after this is fixed https://github.com/jsdoc3/jsdoc/issues/629
+     */
+    _orientation: {value: null},
+
+    /**
+     * Orientation of the bar will always be horizontal by default
+     * can be overridden by setting this property
+     * isn't auto-responsive due to inability to react to parent container dimension changes
+     *
+     * @param {String} orientation
+     * @returns {String}
+     */
+    orientation: {
+        get: function () {
+            return this._orientation;
+        },
+        set: function (orientation) {
+            if (this._orientation !== orientation) {
+                this._orientation = orientation;
+                this.needsDraw = true;
+            }
+        }
+    },
+
+    /**
+     * Redraw based on content or orientation change
+     *
+     * @method
+     */
+    draw: {
+        value: function () {
+            if (this.orientation === "vertical") {
+                this.templateObjects.repetition.element.style.webkitBoxOrient = "vertical";
+                this.templateObjects.repetition.element.style.webkitBoxDirection = "normal";
+                this.templateObjects.repetition.element.style.webkitFlexDirection = "column";
+                this.templateObjects.repetition.element.style.msFlexDirection = "column";
+                this.templateObjects.repetition.element.style.flexDirection = "column";
+            }
+        }
+    }
+});
```

Comment:
BUG (I think): If you set a `dataController` value (legacy name: `contentController` :-), then let the segmented bar draw itself, then set a `data` value (legacy name: `content`), I don't think the segmented bar will correctly update the bindings to the repetition so the new data is displayed (because the bindings are set in templateDidLoad, which is called once and then never again). The most robust way to fix this is to ensure the segmented bar always has an RangeController, either taken or derived from the `dataController` or derived from the `data`. The repetition can then have a binding to `this arrayController` defined in the serialization and the binding doesn't need to change. Setters for `data` and `dataController` will ensure any changes get reflected in a correct `_rangeController`, which will update the repetition. Because you'll have better separation of concerns, the code for this might actually be smaller than the current version. No _data and _dataController properties will be necessary, and It's likely no `needsDraw = 1` will be needed in the `data` and `dataController` setters because the binding will cause changes to `_rangeController` to propagate correctly to the repetition and force a redraw there.